lib/ansible/galaxy/collection.py

```diff
@@ -356,6 +378,45 @@ def from_name(collection, apis, requirement, force, parent=None):
         return req
 
 
+def get_available_api_versions(galaxy_api):
+    headers = {}
+    headers.update(galaxy_api._auth_header(required=False))
+
+    url = _urljoin(galaxy_api.api_server, "api")
+    try:
+        return_data = open_url(url, headers=headers, validate_certs=galaxy_api.validate_certs)
+    except urllib_error.HTTPError as err:
+        if err.code != 401:
+            _handle_http_error(err, galaxy_api, {},
+                               "Error when finding available api versions from %s (%s)" %
+                               (galaxy_api.name, galaxy_api.api_server))
+
+        # assume this is v3 and auth is required.
+        headers = {}
+        headers.update(galaxy_api._auth_header(token_type='Bearer', required=True))
+        # try again with auth
+        try:
+            return_data = open_url(url, headers=headers, validate_certs=galaxy_api.validate_certs)
+        except urllib_error.HTTPError as authed_err:
+            _handle_http_error(authed_err, galaxy_api, {},
+                               "Error when finding available api versions from %s using auth (%s)" %
+                               (galaxy_api.name, galaxy_api.api_server))
+
+    except Exception as e:
+        raise AnsibleError("Failed to get data from the API server (%s): %s " % (url, to_native(e)))
+
+    try:
+        data = json.loads(to_text(return_data.read(), errors='surrogate_or_strict'))
```

Comment:
if `return_data` is the response from `open_url`, you should be able to do `json.load(return_data)` and not worry about reading from the io reader and converting to text